modules/1_prepare/prepare.tf

```diff
@@ -299,7 +299,12 @@ resource "null_resource" "enable_repos" {
 if ( [[ -z "${var.rhel_subscription_username}" ]] || [[ "${var.rhel_subscription_username}" == "<subscription-id>" ]] ) && [[ -z "${var.rhel_subscription_org}" ]]; then
   sudo yum install -y epel-release
 else
-  sudo subscription-manager repos --enable ${var.ansible_repo_name}
+  os_ver=$(cat /etc/os-release | egrep "^VERSION_ID=" | awk -F'"' '{print $2}')
+  if [[ $os_ver != "9"* ]]; then
+    sudo subscription-manager repos --enable ${var.ansible_repo_name}
+  else
+    sudo yum install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-9.noarch.rpm
```

Comment:
Is this related to RHEL or only specific to CentOS? If the answer is yes then you need to check this in above if condition.

Reply:
this is for RHEL 9 only.  For CentOS 9, it uses the same way as CentOS 8 to install epel repo.